students/jeremy_m/lesson04_exercises/mailroom_part_2.py

```diff
@@ -0,0 +1,156 @@
+                             "Thank you so much for your generous donation of:\n"
+                             "{:>21}{:,}\n"
+                             "We will always remember your money fondly."),
+            'letters_confirmation': 'Is this ok?\nType y or n:'
+            }
+
+
+def letters_for_all():
+    """ This function creates a directory in the cwd and fills that with an
+        individual file (letter) for each donor.
+    """
+
+    clear_screen()
+    print('Note: This operation will create a directory in the current working directory')
+    if get_user_input(messages['letters_confirmation']).lower() == 'y':
+        dir_path = os.getcwd() + '/letters'
+        os.mkdir(dir_path)
+        os.chdir(dir_path)
+
+        clear_screen()
+        for donor in donors:
+            donor_filename = '_'.join(donor.split(' ')) + '.txt'
+            with open(donor_filename, 'w+') as new_file:
+                new_file.write(messages['donor_letter'].format(
+                    donor, '$', int(donors[donor][-1])))
+        else:
+            print("Letters Created at:\n{}\n\n".format(dir_path))
+
+
+def send_a_thank_you():
+    """ Gives a user the option to list all donors, or else will create a new
+        donor using the inputted name and adds their new donation using the
+        inputted amount """
+    clear_screen()
+
+    # Ask user for the donors name
+    thank_you_input = get_user_input(messages['thanks'])
+
+    # If the donor exists in donor list ask for latest donation amount and add it
+    if thank_you_input == 'list':
+        clear_screen()
+        print("All donors:")
+        for donor, donations in donors.items():
+            print('\t' + donor)
+
+        thank_you_input = get_user_input('\n\n' + messages['thanks'])
+
+    # If donor isn't in donor list ask for their first donation amount and add it
+    donation_amount = input(
+        'How much did {} contribute?\n---->'.format(thank_you_input))
+
+    if thank_you_input in donors.keys():
+        donors[thank_you_input].append(donation_amount)
+    else:
+        donors[thank_you_input] = []
+        donors[thank_you_input].append(donation_amount)
+
+    clear_screen()
+    # Print a formatted message with donors name and recent donation amount
+    print(messages['thank_you_message'].format('Thank you so much',
+                                               thank_you_input, '$', int(donors[thank_you_input][-1])))
+
+
+def create_a_report():
+    """ Prints a formatted report showing donor name, total donations, number of
+        donations, and average donation amount. """
+    clear_screen()
+
+    # This prints the top of the table
+    print("{:25} | {:12} | {:<19} | {:<16}".format(
+        'Name', 'Total Given', 'Number of Donations', 'Average Donation'))
+    print('-' * 81)
+
+    # Get the sum of a donors donations and add that with their name to a new dict
+    for donor, donations in donors.items():
+        placeholder = 0
```

Comment:
`placeholder` isn't really a good variable name as it doesn't describe what it is.